addons/event/models/event_ticket.py

```diff
@@ -137,5 +139,5 @@ def _get_ticket_multiline_description(self):
         information. """
         return '%s\n%s' % (self.display_name, self.event_id.display_name)
 
-    def _get_ticket_tz(self):
-        return self.event_id.date_tz or self.env.user.tz
+    def _set_tz_context(self):
+        return self.with_context(tz=self.event_id.date_tz or 'UTC' if self.event_id else self.env.user.tz)
```

Comment:
Wouldn't it be better to use `self.with_context(self.event_id.date_tz or self.env.user.tz)`
as environment user UTC should always be set? (and should be UTC for master user?)

Right now there are a bit too many fallbacks.

Reply:
haha ... my thoughts: if there is an event, take the tz of the event (either defined, either UTC as it is its default). If no event, current user TZ is better even if should not happen.